src/base58.cpp

```diff
@@ -259,6 +285,34 @@ CTxDestination DecodeDestination(const std::string& str, const CChainParams& par
             return CScriptID(hash);
         }
     }
+    auto bech = bech32::Decode(str);
+    if (bech.second.size() > 0 && bech.first == params.Bech32HRP()) {
+        // Bech32 decoding
+        int version = bech.second[0];
+        if (ConvertBits<5, 8, false>(data, bech.second.begin() + 1, bech.second.end())) {
```

Comment:
Not obvious to me here why it's `bech.second.begin() + 1`.

Reply:
Added a comment.